mcnp2cad.cpp

```diff
@@ -785,24 +785,53 @@ entity_collection_t GeometryContext::defineCell(  CellCard& cell,  bool defineEm
     case CellCard::SURFNUM:
       {      
         int surface = token.second;
-        bool pos = true;
+        bool positive = true;
         if( surface < 0){
-          pos = false; surface = -surface;
+          positive = false; surface = -surface;
         }
         try{
           SurfaceVolume& surf = makeSurface( deck.lookup_surface_card( surface ) );
-          iBase_EntityHandle surf_handle = surf.define( pos, igm, world_size );
+          iBase_EntityHandle surf_handle = surf.define( positive, igm, world_size );
           stack.push_back(surf_handle);
         }
         catch(std::runtime_error& e) { std::cerr << e.what() << std::endl; }
       }
       break;
     case CellCard::MBODYFACET:
       {
-        int cellnum = std::abs(token.second) / 10;
-        int facet = std::abs(token.second) - (cellnum*10);
-        std::cerr << "Attempting to define facet " << facet << " of cell " << cellnum << std::endl;
-        throw std::runtime_error( "Macrobody facets are not yet supported by mcnp2cad." );
+        int identifier = -std::abs( token.second );
+        int surfacenum = -identifier / 10;
+        int facet = -identifier - ( surfacenum * 10 );
+
+        try{
+          makeSurface( deck.lookup_surface_card( identifier ), NULL, facet);
```

Comment:
I see.  When I was going through the makeSurface function when starting out, I saw if( cace.contains( card ) ) part, and thought that you made the card on your first call to makeSurface (which you do), but you had to call it again to actually make the surface (which you don't).  With the code I have there, it won't work without the first call to makeSurface, but you can capture the first one instead and get rid of the second one.